src/tests/reader.rs

```diff
@@ -62,9 +63,7 @@ fn open_db_with_wrong_distance() {
 
     let rtxn = handle.env.read_txn().unwrap();
     let reader = Reader::<Manhattan>::open(&rtxn, 0, handle.database).unwrap();
-    // TODO: This should fail
-    let ret = reader.nns_by_vector(&rtxn, &[1.0, 2.0], 5, None).unwrap();
-    insta::assert_display_snapshot!(NnsRes(Some(ret)), @"");
+    reader.nns_by_vector(&rtxn, &[1.0, 2.0], 5, None).unwrap();
```

Comment:
```suggestion
    let err = reader.nns_by_vector(&rtxn, &[1.0, 2.0], 5, None).unwrap_err();
    insta::assert_display_snapshot!(err, @"");
```

It's probably better to assert that we got the right error message than to panic 😅